tests/test_ls.py

```diff
@@ -55,7 +55,7 @@ def test_ls_stale_cache_entry(
     lpath = str(random_file)
     rpath = f"{repository}/{temp_branch}/data/{random_file.name}"
 
-    fs.put_file(lpath, rpath)
+    fs.put_file(lpath, rpath, precheck=False)
```

Comment:
This is necessary now since otherwise on the checksum pre-check, `info` is called on a non-existent file, which causes the fall-through and thus another `list_objects` call.

Took me a while to figure out. Verifying the number of expected API counts is nice, but does not tell you where stuff goes wrong unfortunately.

Reply:
Correct.